rpcs3/rpcs3qt/gl_gs_frame.cpp

```diff
@@ -29,13 +31,16 @@ draw_context_t gl_gs_frame::make_context()
 
 	if (m_primary_context)
 	{
-		auto surface = new QOffscreenSurface();
-		surface->setFormat(m_format);
-		surface->create();
+		atomic_t<QOffscreenSurface*> surface = new QOffscreenSurface();
```

Comment:
It doesn't have to be atomic, but it doesn't hurt I guess.

Reply:
Interesting. How do you synchronize access without atomic in C++?

Reply:
This is not cuncurrent access, cuncurrency is what std::atomic (or atomic_t here) is for.

Reply:
OK, I thought that `surface` is used in two threads: the one calling `make_context` and the main thread. I guess I don't understand it correctly.

Reply:
It is used by 2 threads, but cannot at the same time. That is the difference.

Reply:
It doesn't make sense to create an atomic here if it's not used anywhere else

Reply:
I removed it.